transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java

```diff
@@ -230,11 +232,16 @@ private void doReleaseChannel(Channel channel, Promise<Void> promise) {
                          promise);
         } else {
             try {
-                if (offerChannel(channel)) {
-                    handler.channelReleased(channel);
-                    promise.setSuccess(null);
+                final Future<Boolean> f = healthCheck.isHealthy(channel);
```

Comment:
@Scottmitch sure, will do. What do you think the default behavior should be? Check health on release by default or do not check by default?